drift_dev/test/linter_test/linter_test.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:path/path.dart' as p;
+
+void main() {
+  // Test the linter_test.dart file
+  test('linter', () async {
+    final workingDir = p.join(p.current, 'test/linter_test/pkg');
+    expect(
+        await Process.run('dart', ['pub', 'get'], workingDirectory: workingDir)
+            .then((v) => v.exitCode),
+        0);
+    expect(
```

Comment:
We'll also need examples that do have lints, right?

I would suggest:

1. Having testcases defined as strings here
2. Then creating a temporary directory to run `custom_lint` on test code. That directory can be created using `package:test_descriptor`.
3. Instead of running `pub get` which is quite slow, you can make use of the fact that the dependencies including `custom_lint` have already been resolved when tests are running an create a stub package config like this: https://github.com/simolus3/drift/blob/d7cc547836cdce038f76395908e137afd96cbc96/drift_dev/test/analysis/test_utils.dart#L115-L130 That config needs to be in `.dart_tool/package_config.json` of the temporary directory - that should allow running `custom_lint`.

Let me know if there's anything I can help with here

Reply:
`// expect_lint: drift_build_errors`
Will throw a lint error is an error is NOT found. It's tests the lint.

Reply:
It needs to hold the built code too. If we do it with strings, it will look like a mess:
```dart
final _db = """
import 'package:drift/drift.dart';

part 'db.g.dart';

class Users extends Table {
  late final id = integer().autoIncrement()();
  late final name = text()();
  // expect_lint: drift_build_errors
  late final age = integer();
}

class BrokenTable extends Table {
  // expect_lint: drift_build_errors
  IntColumn get unknownRef => integer().customConstraint('CHECK foo > 10')();
}

@DriftDatabase(tables: [Users])
class TestDatabase extends _\$TestDatabase {
  TestDatabase(super.e);

  @override
  int get schemaVersion => 1;

  a() async {
    transaction(
      () async {
        // expect_lint: unawaited_futures_in_transaction
        into(users).insert(UsersCompanion.insert(name: 'name'));
      },
    );
  }
}
""" 
final _dbPart = """

// GENERATED CODE - DO NOT MODIFY BY HAND

part of 'db.dart';

// ignore_for_file: type=lint
class $UsersTable extends Users with TableInfo<$UsersTable, User> {
  @override
  final GeneratedDatabase attachedDatabase;
  final String? _alias;
  $UsersTable(this.attachedDatabase, [this._alias]);
  static const VerificationMeta _idMeta = const VerificationMeta('id');
  @override
  late final GeneratedColumn<int> id = GeneratedColumn<int>(
      'id', aliasedName, false,
      hasAutoIncrement: true,
      type: DriftSqlType.int,
      requiredDuringInsert: false,
      defaultConstraints:
          GeneratedColumn.constraintIsAlways('PRIMARY KEY AUTOINCREMENT'));
  static const VerificationMeta _nameMeta = const VerificationMeta('name');
  @override
  late final GeneratedColumn<String> name = GeneratedColumn<String>(
      'name', aliasedName, false,
      type: DriftSqlType.string, requiredDuringInsert: true);
  @override
  List<GeneratedColumn> get $columns => [id, name];
  @override
  String get aliasedName => _alias ?? actualTableName;
  @override
  String get actualTableName => $name;
  static const String $name = 'users';
  @override
  VerificationContext validateIntegrity(Insertable<User> instance,
      {bool isInserting = false}) {
    final context = VerificationContext();
    final data = instance.toColumns(true);
    if (data.containsKey('id')) {
      context.handle(_idMeta, id.isAcceptableOrUnknown(data['id']!, _idMeta));
    }
    if (data.containsKey('name')) {
      context.handle(
          _nameMeta, name.isAcceptableOrUnknown(data['name']!, _nameMeta));
    } else if (isInserting) {
      context.missing(_nameMeta);
    }
    return context;
  }

  @override
  Set<GeneratedColumn> get $primaryKey => {id};
  @override
  User map(Map<String, dynamic> data, {String? tablePrefix}) {
    final effectivePrefix = tablePrefix != null ? '$tablePrefix.' : '';
    return User(
      id: attachedDatabase.typeMapping
          .read(DriftSqlType.int, data['${effectivePrefix}id'])!,
      name: attachedDatabase.typeMapping
          .read(DriftSqlType.string, data['${effectivePrefix}name'])!,
    );
  }


/// And much much more
....
```

Reply:
Let me know if you still want me to have the tests in strings.

Reply:
A separate package / directory to hold test examples is fine in that case :+1: